python/GafferUI/PathListingWidget.py

```diff
@@ -800,24 +914,50 @@ def drawBranches( self, painter, rect, index ) :
 		# rest to Qt.
 		painter.fillRect( rect, QtGui.QColor( *(_styleColors["backgroundRaised"]) ) )
 
-		self.__drawSelectionHighlight( painter, rect, index )
+		self.__drawSelectionHighlight( painter, rect, index, fullRow = True )
 
 		QtWidgets.QTreeView.drawBranches( self, painter, rect, index )
 
-	def __drawSelectionHighlight( self, painter, rect, index ) :
+	def __drawSelectionHighlight( self, painter, rect, index, fullRow = False ) :
```

Comment:
I haven't followed the new `fullRow` logic here fully, but something is off :

![image](https://user-images.githubusercontent.com/1133871/177303392-28078da0-3dcb-4e7e-a15c-199b21b295e4.png)

Both `/group` and `/group/spotLight` are selected, but the descendant selection is clobbering the full selection for all of the row except the branches section.

Reply:
My logic behind `fullRow` is that when drawing the branch, we shouldn't iterate over the columns because the `rect` given to us covers only the left-most width of the branch area.

If we do iterate over columns when drawing the branch area and the last column is selected, the `width` calculated will be negative (because `rect.right()` is the width of the branch area) and the whole row (minus the branch area) is drawn. 

In theory we could also use it for drawing in `Row` and `Rows` modes, but I believe that would require a new method like `selectionMode()` or similar for `_TreeView` to be able to know what mode it's in. I avoided that to keep the overall changes down.

I'm open to clearer naming or better logic of course.

Reply:
Fixing the highlight behavior made it a bit easier to organize the branch vs. row data logic, so the `fullRow` bit is no longer needed. Fixed in b318f952c56fffaa60bcd2eb0c92ae1b6e35cc29